modules/core/src/main/scala/predicate.scala

```diff
@@ -14,16 +14,55 @@ import cats.implicits._
 import Query._
 import QueryInterpreter.mkErrorResult
 
+/**
+ * A reified function over a `Cursor`.
+ *
+ * Query interpreters will typically need to introspect predicates (eg. in the doobie module
+ * we need to be able to construct where clauses from predicates over fields/attributes), so
+ * these cannot be arbitrary functions `Cursor => Boolean`.
+ */
+trait Term[T] extends Product with Serializable {
+  def apply(c: Cursor): Result[T]
+}
+
+sealed trait Path {
+  def path: List[String]
+  def extend(prefix: List[String]): Path
```

Comment:
Reading the method name I expected it to take a suffix. Would something like `prefix` be a better name for this?

Reply:
It's really "extend towards the root". Could you live with `prepend`?

Reply:
Yes, I think `prepend` would be less confusing. To me anyway :-)